consensus/polybft/contractsapi/contractsapi.go

```diff
@@ -0,0 +1,285 @@
+}
+
+type Uptime struct {
+	EpochID     *big.Int      `abi:"epochId"`
+	UptimeData  []*UptimeData `abi:"uptimeData"`
+	TotalBlocks *big.Int      `abi:"totalBlocks"`
+}
+
+func (u *Uptime) EncodeAbi() ([]byte, error) {
+	return UptimeABIType.Encode(u)
+}
+
+func (u *Uptime) DecodeAbi(buf []byte) error {
+	return decodeStruct(UptimeABIType, buf, &u)
+}
+
+type CommitEpoch struct {
+	ID     *big.Int `abi:"id"`
+	Epoch  *Epoch   `abi:"epoch"`
+	Uptime *Uptime  `abi:"uptime"`
+}
+
+func (c *CommitEpoch) EncodeAbi() ([]byte, error) {
+	return ChildValidatorSet.Abi.Methods["commitEpoch"].Encode(c)
+}
+
+func (c *CommitEpoch) DecodeAbi(buf []byte) error {
+	return decodeMethod(ChildValidatorSet.Abi.Methods["commitEpoch"], buf, c)
+}
+
+type StateSenderContractImpl struct {
+	Artifact *artifact.Artifact
+
+	SyncState SyncState
+}
+
+type SyncState struct {
+	Receiver types.Address `abi:"receiver"`
+	Data     []byte        `abi:"data"`
+}
+
+func (s *SyncState) EncodeAbi() ([]byte, error) {
+	return StateSender.Abi.Methods["syncState"].Encode(s)
+}
+
+func (s *SyncState) DecodeAbi(buf []byte) error {
+	return decodeMethod(StateSender.Abi.Methods["syncState"], buf, s)
+}
+
+var (
+	StateSyncedEventType = abi.MustNewEvent("event StateSynced(uint256 indexed id,address indexed sender,address indexed receiver,bytes data)") //nolint:all
+)
+
+type StateSyncedEvent struct {
+	ID       *big.Int      `abi:"id"`
+	Sender   types.Address `abi:"sender"`
+	Receiver types.Address `abi:"receiver"`
+	Data     []byte        `abi:"data"`
+}
+
+func (S *StateSyncedEvent) ParseLog(log *ethgo.Log) error {
+	return decodeEvent(StateSyncedEventType, log, S)
+}
+
+type CheckpointManagerContractImpl struct {
+	Artifact *artifact.Artifact
+
+	Submit Submit
+}
+
+var CheckpointMetadataABIType = abi.MustNewType("tuple(bytes32 blockHash,uint256 blockRound,bytes32 currentValidatorSetHash)")
+
+type CheckpointMetadata struct {
+	BlockHash               types.Hash `abi:"blockHash"`
+	BlockRound              *big.Int   `abi:"blockRound"`
+	CurrentValidatorSetHash types.Hash `abi:"currentValidatorSetHash"`
+}
+
+func (c *CheckpointMetadata) EncodeAbi() ([]byte, error) {
+	return CheckpointMetadataABIType.Encode(c)
+}
+
+func (c *CheckpointMetadata) DecodeAbi(buf []byte) error {
+	return decodeStruct(CheckpointMetadataABIType, buf, &c)
+}
+
+var CheckpointABIType = abi.MustNewType("tuple(uint256 epoch,uint256 blockNumber,bytes32 eventRoot)")
+
+type Checkpoint struct {
+	Epoch       *big.Int   `abi:"epoch"`
+	BlockNumber *big.Int   `abi:"blockNumber"`
+	EventRoot   types.Hash `abi:"eventRoot"`
+}
+
+func (c *Checkpoint) EncodeAbi() ([]byte, error) {
+	return CheckpointABIType.Encode(c)
+}
+
+func (c *Checkpoint) DecodeAbi(buf []byte) error {
+	return decodeStruct(CheckpointABIType, buf, &c)
+}
+
+var NewValidatorSetABIType = abi.MustNewType("tuple(address _address,uint256[4] blsKey,uint256 votingPower)")
+
+type NewValidatorSet struct {
+	_address    types.Address `abi:"_address"`
```

Comment:
We need to handle this

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1062/commits/d47f9485d8bba2ca81d07006c241fd7b392ff0f7